### DISCUSSION TITLE
How to implement FilesystemDataset

### DISCUSSION THREAD

**Discussion by patcharees** [mlflow/mlflow#12578] [Q&A]
I would like to log my dataset from normal file system on a local machine. But I see that the FilesystemDataset class has not been implemented.

Now I am running mlflow server from pip install, starting server by issue this command
`mlflow server`

How can I implement this class, and use it as a lib to log dataset on another project?

Thanks in advance

**Comment by B-Step62** [comment]
@patcharees The concrete dataset classes represent a particular the data type e.g. `NumpyDataset` for Numpy array, `PandasDataset` for Pandas dataframe. How/where the data is loaded is modeled as `DataSource`. So for your use case:
1. Check if the data type of your dataset has concrete `Dataset` class in MLflow. If not, it would be awesome if you can contribute.
2. Implement custom class that inherits [FileSystemDatasetSource](https://github.com/mlflow/mlflow/blob/e6c7d6e0362c92d6145d59b83534162cb0e8913f/mlflow/data/filesystem_dataset_source.py#L7C7-L7C30) class. Basically you need to override all methods marked with `@abstractmethod`. 

You can also refer to how existing concrete classes are implemented: https://mlflow.org/docs/latest/tracking/data-api.html#dataset
